compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1876,7 +1876,8 @@ private void handleDeclaredWithVar(BLangVariable variable, AnalyzerData data) {
                 handleWildCardBindingVariable(simpleVariable, currentEnv);
 
                 long ownerSymTag = currentEnv.scope.owner.tag;
-                if ((ownerSymTag & SymTag.INVOKABLE) == SymTag.INVOKABLE || (ownerSymTag & SymTag.LET) == SymTag.LET) {
+                if ((ownerSymTag & SymTag.INVOKABLE) == SymTag.INVOKABLE || (ownerSymTag & SymTag.LET) == SymTag.LET ||
+                        (ownerSymTag & SymTag.PACKAGE) == SymTag.PACKAGE) {
```

Comment:
If the variable is in the package scope, we define those variables in the symbol enter. Here, what we define are variables within functions, and the variables' owner becomes let. In the following scenario, the owner tag should also be equal to let. 

In the current implementation, we set the owner as `package` instead of `let`. This is the root cause of the issue.

```ballerina
type Student record {|
    string fname;
    string lname;
    int id;
|};

Student[] students = [];

var v1 = from var st in students
    let var name = st.fname + st.lname
    select name;

function testModuleLevelLetClause() {
    var v2 = from var st in students
        let var name = st.fname + st.lname
        select name;
}
```